relay-server/src/actors/store.rs

```diff
@@ -199,51 +199,122 @@ impl StoreService {
             }
         }
 
+        let remote_addr = envelope.meta().client_addr().map(|addr| addr.to_string());
+
+        let kafkamessages = self.extract_kafka_messages(
+            event_item,
+            event_id,
+            scoping,
+            start_time,
+            remote_addr,
+            &mut attachments,
+        );
+
+        for message in kafkamessages {
+            if let Ok(()) = self.produce(topic, scoping.organization_id, &message) {
+                // If message was succesfully sent, make the right log message.
+                match message {
+                    KafkaMessage::Attachment(_) => {
+                        metric!(
+                            counter(RelayCounters::ProcessingMessageProduced) += 1,
+                            event_type = "attachment"
+                        );
+                    }
+                    KafkaMessage::Event(_) => metric!(
+                        counter(RelayCounters::ProcessingMessageProduced) += 1,
+                        // The unwrap is safe because in self.extract_kafka_messages() it only
+                        // pushes an event message if the event_item is_some()..
+                        // still I would prefer to not have any unwraps.
+                        event_type = &(event_item.unwrap().ty().to_string())
+                    ),
+                    _ => {}
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn extract_kafka_messages(
+        &self,
+        event_item: Option<&Item>,
+        event_id: Option<EventId>,
+        scoping: Scoping,
+        start_time: Instant,
+        remote_addr: Option<String>,
+        attachments: &mut Vec<ChunkedAttachment>,
+    ) -> Vec<KafkaMessage> {
+        let mut kafkamsgs = vec![];
         if let Some(event_item) = event_item {
-            relay_log::trace!("Sending event item of envelope to kafka");
+            if matches!(event_item.ty(), ItemType::Transaction) {
+                self.extract_kafka_attachments(
+                    &mut kafkamsgs,
+                    attachments.clone(),
```

Comment:
You do need this `clone` here.

The function can own the `attachments`, but changing the signature to
```rust 
....
mut attachments: Vec<ChunkedAttachment>
```